evaluation/evaluate.py

```diff
@@ -259,52 +270,68 @@ def plot_diagrams(questions_with_ratings: list, requested_metrics: list, passing
         "mean_rating": "Rating Score",
     }
 
-    stats_y_lim = {"pass_count": len(questions_with_ratings), "pass_rate": 1.0, "mean_rating": 5.0}
-
+    stats_y_lim = {
+        "pass_count": len(questions_with_ratings_dict[next(iter(questions_with_ratings_dict))]),
+        "pass_rate": 1.0,
+        "mean_rating": 5.0,
+    }
     # Draw the chart for the results
-    data = [data for _, data in rating_stat_data.items()]
-    titles = [display_stats_name[mn] for mn in rating_stat_data.keys()]
-    y_labels = [stats_y_labels[mn] for mn in rating_stat_data.keys()]
-    y_lims = [stats_y_lim[mn] for mn in rating_stat_data.keys()]
-    layout = (int(np.ceil(len(rating_stat_data) / 3)), 3 if len(rating_stat_data) > 3 else len(rating_stat_data))
+    data = {}
+    for key in questions_with_ratings_dict:
+        data_per_model = [data for _, data in rating_stat_data[key].items()]
+        data[key] = data_per_model
+        titles = [display_stats_name[mn] for mn in rating_stat_data[key].keys()]
+        y_labels = [stats_y_labels[mn] for mn in rating_stat_data[key].keys()]
+        y_lims = [stats_y_lim[mn] for mn in rating_stat_data[key].keys()]
+        layout = (
+            int(np.ceil(len(rating_stat_data[key]) / 3)),
+            3 if len(rating_stat_data[key]) > 3 else len(rating_stat_data[key]),
+        )
 
     plot_bar_charts(
-        layout, data, titles, y_labels, results_dir / "evaluation_results.pdf", y_max_lim=y_lims, width=width
+        layout,
+        data,
```

Comment:
> BTW, the above crashes with 2 models as well, so I'll go on to assume that it was only ever supposed to work for multiples of 3 or sth.

That's weird since everything worked properly on my side, no matter single or multiples

Reply:
Yeah, I'm sure it works as expected for 1 or 3 models, it's for 2 or >3 that it breaks.

Reply:
In the code I've pushed I have integrated the newest models we support. Please try and have this be dynamic!